src/playback-watcher.js

```diff
@@ -90,6 +92,22 @@ export default class PlaybackWatcher {
     const cancelTimerHandler = () => this.cancelTimer_();
     const fixesBadSeeksHandler = () => this.fixesBadSeeks_();
 
+    const mpc = this.masterPlaylistController_;
+
+    const loaderTypes = ['main', 'subtitle', 'audio'];
+    const loaderChecks = {};
+
+    loaderTypes.forEach((type) => {
+      loaderChecks[type] = {
+        reset: () => this.resetSegmentDownloads_(type),
+        updateend: () => this.checkSegmentDownloads_(type)
+      };
+
+      mpc[`${type}SegmentLoader_`].on('updateend', loaderChecks[type].updateend);
+      mpc[`${type}SegmentLoader_`].on('playlist-update', loaderChecks[type].reset);
+      this.tech_.on(['seeked', 'seeking'], loaderChecks[type].reset);
```

Comment:
Playback stalls should not be detected right after seeking. This prevents one segment playlists (such as a single vtt file for a video) from being detected as stalling as the buffer will not change, since it will be the entire video duration.